### PR TITLE
Add tooltip to contacts

### PR DESCRIPTION
**Pull request by muzakparov**
Related to #8790 

Contact page:

contact

Seed phrase page:

seed

Not sure if it is necessary, but I think that copy logic could be abstracted to hooks since it seems to repeat in different places.

### CODE REVIEW

File: ui/app/components/ui/export-text-container/export-text-container.component.js [hunk 1]
```diff
@@ -5,25 +5,33 @@ import { exportAsFile } from '../../../helpers/utils/util'
...
-          </div>
+          <div className="export-text-container__text notranslate">{text}</div>
         </div>
         <div className="export-text-container__buttons-container">
           <div
             className="export-text-container__button export-text-container__button--copy"
-            onClick={() => copyToClipboard(text)}
+            onClick={() => {
+              this.setState({ copied: true })
+              setTimeout(() => this.setState({ copied: false }), 3000)
```
Comment (Gudahtt) [inline ui/app/components/ui/export-text-container/export-text-container.component.js:26]: This will throw an error in the console if this component is unmounted when this timer resolves. It's a harmless error in this case, but still nice to avoid. It could be avoided by saving the timeout ID, and clearing the timeout on unmount.

File: ui/app/pages/settings/contact-list-tab/view-contact/view-contact.component.js [hunk 2]
```diff
@@ -48,28 +66,41 @@ export default class ViewContact extends PureComponent {
...
+                    ? t('copiedExclamation')
+                    : t('copyToClipboard')
+                }
               >
-                <Copy size={20} color="#3098DC" />
-              </button>
+                <button
+                  className="address-book__view-contact__group__static-address--copy-icon"
+                  onClick={() => {
+                    const { checkSummedAddress } = this.props
```
Comment (Gudahtt) [inline ui/app/pages/settings/contact-list-tab/view-contact/view-contact.component.js:87]: Nit: It would be better to avoid shadowed variables if possible. In this case there's no need to declare another `checkSummedAddress` variable here, as there's already one in-scope.

```suggestion
```

File: ui/app/components/ui/export-text-container/export-text-container.component.js (hunk 1 above)
Comment (Gudahtt) [reply ui/app/components/ui/export-text-container/export-text-container.component.js:26]: The same applies to the similar change made to the `ViewContact` component as well

File: ui/app/pages/settings/contact-list-tab/view-contact/view-contact.component.js [hunk 3]
```diff
@@ -48,28 +66,41 @@ export default class ViewContact extends PureComponent {
...
-              >
-                { quadSplit(checkSummedAddress) }
+              <div className="address-book__view-contact__group__static-address">
+                {quadSplit(checkSummedAddress)}
               </div>
-              <button
-                className="address-book__view-contact__group__static-address--copy-icon"
-                onClick={() => copyToClipboard(checkSummedAddress)}
+              <Tooltip
+                wrapperClassName="address-book__tooltip-wrapper"
```
Comment (Gudahtt) [inline ui/app/pages/settings/contact-list-tab/view-contact/view-contact.component.js:76]: It looks like this class doesn't exist? :thinking: If so, better to not reference it.
```suggestion
```

File: ui/app/pages/settings/contact-list-tab/index.scss [hunk 4]
```diff
@@ -21,7 +21,7 @@
     border-bottom: 1px solid #dedede;
 
     &:hover {
-      border: 1px solid #037DD6;
```
Comment (Gudahtt) [inline ui/app/pages/settings/contact-list-tab/index.scss:24]: A lot of these changes seem unrelated to the primary purpose of this PR. Is this the work of some auto-formatter or something? :thinking: 

This particular change (lower-casing hex colours) is in-line with our current stylelint config (which is broken but will be fixed shortly), but some of the other changes (e.g. adding a leading zero to decimal numbers) are in conflict with it.

**Review by Gudahtt — COMMENTED** [review]
Thanks @muzakparov! Looks pretty good overall - I had a few comments/questions.

>Not sure if it is necessary, but I think that copy logic could be abstracted to hooks since it seems to repeat in different places.

Definitely not necessary, but I agree that would be great. We'd just need to convert these to functional components first, so we could use hooks.

**Comment by muzakparov** [comment]
thank you @Gudahtt 
refactored to function components and added two react hooks, now it clears timeout on unmount
I am running `yarn lint:fix` before committing my changes

File: ui/app/pages/settings/contact-list-tab/index.scss (hunk 4 above)
Comment (Gudahtt) [reply ui/app/pages/settings/contact-list-tab/index.scss:24]: stylelint has been re-renabled for all auto-fixable rules in #8989, which was just merged. So most of these changes should vanish now if you rebase on `develop`,

**Comment by Gudahtt** [comment]
Hey @muzakparov, we only just merged the PR that included uncommenting that linting rule. After you update this branch, it should work properly if you try linting again. Sorry for the confusion!

**Comment by muzakparov** [comment]
Cool stuff! I merged with your changes and seems to be fine now. Also, github seems to be having some synchronization issues at the moment, so I am unable to see my changes on the Pull Request.

**Comment by muzakparov** [comment]
Linting seems a bit different on CI/CD, I had to add a comma to make the linting test pass.

**Comment by Gudahtt** [comment]
Oh, that might be because we recently updated `eslint` in #8978, which resulted in `eslint` requiring dangling commas for parameters as well. You might need to run `yarn` locally to get the updated version, or you might be running a different version of `eslint` that is installed globally or in your editor.

**Comment by muzakparov** [comment]
I merged with develop branch, didn't notice that I had many commits sorry

**Comment by muzakparov** [comment]
@Gudahtt could you please help me with the error:)

**Comment by Gudahtt** [comment]
Sure, no problem! I'll take a look

**Comment by Gudahtt** [comment]
I'm not sure exactly what went wrong, but I backtracked and merged `develop` into this branch again, and now it seems to be OK!

Sorry for the delay in reviewing this - I'm hoping to get to it today or tomorrow.

File: ui/app/hooks/useTimeout.js [hunk 5]
```diff
@@ -0,0 +1,40 @@
+import { useState, useEffect, useRef, useCallback } from 'react'
+
+/**
+ * useTimeout
+ *
+ * @param {Function}  cb   - callback function inside setTimeout
+ * @param {number}  delay   - delay in ms
+ * @param {boolean}  immediate   - delay in ms
```
Comment (Gudahtt) [inline ui/app/hooks/useTimeout.js:8]: This should be marked as optional, and the description should be updated. Something like this perhaps:
```suggestion
 * @param {boolean}  [immediate] - determines whether the timeout is invoked immediately
```

File: ui/app/hooks/useTimeout.js [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+      setTriggered(false)
+      saveCb.current()
+    }, delay)
+
+    return () => {
+      clearTimeout(id)
+    }
+  }, [delay, triggered])
+
+  const trigger = useCallback(() => {
```
Comment (Gudahtt) [inline ui/app/hooks/useTimeout.js:35]: Nit: I was a bit confused by this name at first; could this be changed to something like `startTimeout`?

File: ui/app/hooks/useTimeout.js [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+    setTriggered(true)
```
Comment (Gudahtt) [inline ui/app/hooks/useTimeout.js:36]: Note that this wouldn't reset the timer if the user copied the address multiple times. This could lead to strange behavior, e.g. the text rapidly flashing between states, rather than showing for 3 seconds.

Perhaps you could store the timeout id in state instead of this `triggered` boolean? Then the old timeout could be cleared here (if it's set), allowing the timer to be reset.

**Review by Gudahtt — COMMENTED** [review]
These changes look fantastic! I had a couple of follow up suggestions/questions, let me know what you think.

File: ui/app/hooks/useTimeout.js (hunk 7 above)
Comment (muzakparov) [reply ui/app/hooks/useTimeout.js]: thank you for the suggestions, now using `timeoutId` in a state, so that it is cleared/reset whenever `startTimeout` is called

**Review by Gudahtt — APPROVED** [review]
LGTM, great work!
